### PR TITLE
Add new data source batch_compute_environment

### PR DESCRIPTION
**Pull request by gheine**

### CODE REVIEW

File: aws/data_source_aws_batch_compute_environment_test.go [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform/helper/acctest"
+	"github.com/hashicorp/terraform/helper/resource"
+)
+
+func TestAccAWSBatchDataSource_ecsCluster(t *testing.T) {
```
Comment (bflad) [inline aws/data_source_aws_batch_compute_environment_test.go]: Minor naming nitpick: I'd rename this to `TestAccAWSBatchComputeEnvironmentDataSource_basic`

File: aws/data_source_aws_batch_compute_environment_test.go [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+    ]
+    type = "EC2"
+  }
+  service_role = "${aws_iam_role.aws_batch_service_role.arn}"
+  type = "MANAGED"
+  depends_on = ["aws_iam_role_policy_attachment.aws_batch_service_role"]
+}
+
+data "aws_batch_compute_environment" "default" {
+  compute_environment_name = "${aws_batch_compute_environment.sample.name}"
```
Comment (bflad) [inline aws/data_source_aws_batch_compute_environment_test.go]: This is currently failing testing:
```
=== RUN   TestAccAWSBatchDataSource_ecsCluster
--- FAIL: TestAccAWSBatchDataSource_ecsCluster (3.65s)
    testing.go:518: Step 0 error: Error planning: 1 error(s) occurred:
        
        * data.aws_batch_compute_environment.default: 1 error(s) occurred:
        
        * data.aws_batch_compute_environment.default: Resource 'aws_batch_compute_environment.sample' does not have attribute 'name' for variable 'aws_batch_compute_environment.sample.name'
```

File: aws/data_source_aws_batch_compute_environment.go [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+		ComputeEnvironments: []*string{aws.String(d.Get("compute_environment_name").(string))},
+	}
+	log.Printf("[DEBUG] Reading Batch Compute Environment: %s", params)
+	desc, err := conn.DescribeComputeEnvironments(params)
+
+	if err != nil {
+		return err
+	}
+
+	for _, computeEnvironment := range desc.ComputeEnvironments {
```
[the remark this answers is not in the document]
Comment (gheine) [reply aws/data_source_aws_batch_compute_environment.go]: yes, that's fair. I was basing this on data_source_aws_ecs_cluster.go which would benefit from the same change.

**Comment by gheine** [comment]
I've updated data_source_aws_ecs_cluster accordingly, see #4286

File: aws/data_source_aws_batch_compute_environment_test.go [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+		}
+
+		return nil
+	}
+}
+
+func testAccDataSourceAwsBatchComputeEnvironmentConfig(rName string) string {
+	return fmt.Sprintf(`
+resource "aws_iam_role" "ecs_instance_role" {
+  name = "ecs_instance_role"
```
[the remark this answers is not in the document]
Comment (gheine) [reply aws/data_source_aws_batch_compute_environment_test.go]: Done

**Comment by bflad** [comment]
This has been released in [version 1.16.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1160-april-25-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.

**Pull request merged**